src/goto-programs/goto_program.h

```diff
@@ -87,6 +88,8 @@ class goto_programt
   class instructiont
   {
   public:
+    std::mutex instruction_mutex;
```

Comment:
Is this the right granularity for thread safety?

Reply:
Shall we use std::shared_mutex (reader-writer lock) if instruction objects are read more frequently than they are modified?